backend/danswer/connectors/file/connector.py

```diff
@@ -0,0 +1,124 @@
+import os
+import zipfile
+from collections.abc import Generator
+from enum import Enum
+from pathlib import Path
+from typing import Any
+from typing import IO
+
+from danswer.configs.app_configs import INDEX_BATCH_SIZE
+from danswer.configs.constants import DocumentSource
+from danswer.connectors.file.utils import check_file_ext_is_valid
+from danswer.connectors.file.utils import get_file_ext
+from danswer.connectors.interfaces import GenerateDocumentsOutput
+from danswer.connectors.interfaces import LoadConnector
+from danswer.connectors.models import Document
+from danswer.connectors.models import Section
+from danswer.utils.logging import setup_logger
+
+
+logger = setup_logger()
+
+_LINK_METADATA_FLAG = "#LINK="
+
+
+def _get_files_from_zip(
+    zip_location: str | Path,
+) -> Generator[tuple[str, IO[Any]], None, None]:
+    with zipfile.ZipFile(zip_location, "r") as zip_file:
+        for file_name in zip_file.namelist():
+            with zip_file.open(file_name, "r") as file:
+                yield os.path.basename(file_name), file
+
+
+def _open_files_at_location(
+    file_path: str | Path,
+) -> Generator[tuple[str, IO[Any]], Any, None]:
+    extension = get_file_ext(file_path)
+
+    if extension == ".zip":
```

Comment:
tar.gz, .tar.bz2, rar, maybe 7zip? are common as well. Does it take much effort to support those, like does the zipfile library handle these?

Reply:
I think it's fine for now, can add those as follow ups if needed